### PR TITLE
Add type for server config

### PR DESCRIPTION
**Pull request by matissepage**
warning Depends on #2163 

**This PR is still in progress**

## What does this PR do ?

Add type for server config 

### How should this be manually tested?

```npm run test:unit```
or
```npm run test```

## Other changes

 change the import to make it work

### CODE REVIEW

**Review by Aschen — APPROVED** [review]

**Review by xbill82 — APPROVED** [review]

File: lib/types/index.ts [hunk 1]
```diff
@@ -32,4 +32,5 @@ export * from './Global';
 export * from './config/PasswordPolicy';
 export * from './config/KuzzleConfiguration';
 export * from './config/RoleDefinition';
-export * from './config/ProfileDefinition';
\ No newline at end of file
+export * from './config/ProfileDefinition';
+export * from './config/ServerConfiguration';
```
Comment (alexandrebouthinon) [inline lib/types/index.ts:36]:
```suggestion
export * from './config/ServerConfiguration';

```

**Review by alexandrebouthinon — APPROVED** [review]

File: lib/types/config/ServerConfiguration.ts [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+    *
+    * Currently supported are "combined" (=Apache combined logs format)
+    * and "logstash".
+    *
+    * "logstash" will output the whole request input to JSON, ready to
+    * be consumed by logstash agent.
+    *
+    * @default "combined"
+    */
+    accessLogFormat: string;
```
Comment (Aschen) [inline lib/types/config/ServerConfiguration.ts:60]: This is `"combined" | "logstash"`

File: lib/types/config/ServerConfiguration.ts [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+      * @default true
+      */
+      allowCompression: boolean
+
+      /**
+       * Set to "false" to disable HTTP support
+       * 
+       * @default true
+       */
+        enabled: true
```
Comment (Aschen) [inline lib/types/config/ServerConfiguration.ts:92]: Is this always `true` or `boolean` ? Also you have an indentation problem starting here

File: lib/types/config/ServerConfiguration.ts [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+              * @default 0
+              */
+              rateLimit: number
+
+              /**
+               * Set to "true" to enable realtime notifications like "TokenExpired" notifications
+               * 
+               * @default true
+               */
+                realtimeNotifications: boolean
```
Comment (Aschen) [inline lib/types/config/ServerConfiguration.ts:210]: Wrong indentation

**Review by Aschen — CHANGES_REQUESTED** [review]

**Review by Aschen — APPROVED** [review]

**Pull request merged**